asteroid/data/dns_dataset.py

```diff
@@ -33,28 +37,54 @@ def __len__(self):
     def __getitem__(self, idx):
         """Gets a mixture/sources pair.
         Returns:
-            mixture, vstack([source_arrays])
+            (mixture, clean, noise) if self.load_noise is true, else (mixture, clean)
         """
         utt_info = self.mix_infos[self.wav_ids[idx]]
         # Load mixture
-        x = torch.from_numpy(sf.read(utt_info["mix"], dtype="float32")[0])
+        x_np, sr = sf.read(utt_info["mix"], dtype="float32")
+        assert sr == self.sample_rate
+        start, stop = get_wav_random_start_stop(
+            len(x_np), int(self.segment * self.sample_rate) if self.segment is not None else None
+        )
+        x = torch.from_numpy(x_np[start:stop])
         # Load clean
-        speech = torch.from_numpy(sf.read(utt_info["clean"], dtype="float32")[0])
-        # Load noise
-        noise = torch.from_numpy(sf.read(utt_info["noise"], dtype="float32")[0])
-        return x, speech, noise
+        speech = torch.from_numpy(
+            sf.read(utt_info["clean"], dtype="float32", start=start, stop=stop)[0]
+        )
+        if self.segment is not None:
+            x = torch.nn.functional.pad(
+                x, (0, max(0, int(self.segment * self.sample_rate) - len(x)))
+            )
+            speech = torch.nn.functional.pad(
+                speech, (0, max(0, int(self.segment * self.sample_rate) - len(speech)))
+            )
+        if self.load_noise:
+            # Load noise
+            noise = torch.from_numpy(
+                sf.read(utt_info["noise"], dtype="float32", start=start, stop=stop)[0]
+            )
+            if self.segment is not None:
+                noise = torch.nn.functional.pad(
+                    noise, (0, max(0, int(self.segment * self.sample_rate) - len(noise)))
+                )
+            return x, speech, noise
+        else:
+            return x, speech
 
     def get_infos(self):
         """Get dataset infos (for publishing models).
 
         Returns:
             dict, dataset infos with keys `dataset`, `task` and `licences`.
         """
-        infos = dict()
-        infos["dataset"] = self.dataset_name
-        infos["task"] = "enhancement"
-        infos["licenses"] = [dns_license]
-        return infos
+        return {
+            "dataset": self.dataset_name,
+            "sample_rate": self.sample_rate,
+            "segment": self.segment,
+            "task": "enhancement",
+            "licenses": [dns_license],
+            "n_src": 2 if self.load_noise else 1,
```

Comment:
I'm unsure about that. This will depend on the model. You might need to noise to compute a mask of something but still output only the speech in your model. 
I'd have to check all the places this `n_src` is used to be sure it's ok.

Reply:
Oh I though `n_src` in the dataset info means the number of sources returned by `__getitem__`

Reply:
Actually, none of the datasets return `n_src` in info, so it's a choice we have to make. 
`get_infos` is only used for publishing models for now, and `n_src` is not used, is there specific reason you added it?

Reply:
No, we can leave it out for now